mobile/src/main/java/org/openhab/habdroid/background/NotificationUpdateObserver.java

```diff
@@ -111,13 +111,17 @@ public void onChanged(List<WorkInfo> workInfos) {
                 final String value = data.getString(ItemUpdateWorker.OUTPUT_DATA_VALUE);
                 final boolean hadConnection =
                         data.getBoolean(ItemUpdateWorker.OUTPUT_DATA_HAS_CONNECTION, false);
+                final boolean isDemo =
+                        data.getBoolean(ItemUpdateWorker.OUTPUT_DATA_IS_DEMO, false);
                 final int httpStatus =
                         data.getInt(ItemUpdateWorker.OUTPUT_DATA_HTTP_STATUS, 0);
 
                 retryInfos.add(new BackgroundTasksManager.RetryInfo(entry.first, itemName, value));
                 if (hadConnection) {
                     errors.add(mContext.getString(
                             R.string.item_update_http_error, itemName, httpStatus));
+                } else if (isDemo) {
+                    errors.add(mContext.getString(R.string.item_update_connection_is_demo));
```

Comment:
I'm not sure I like this. This feels like it should be handled on preferences side, by
- disabling the upload category in demo mode
- calling `scheduleWorker()` when demo mode toggles and
- checking for demo mode being enabled there and returning early (like for disabled setting) if it is

Reply:
I implemented it this way, but if there is an error notification, the user can click retry and it's going to be send to the demo server.

Reply:
Hmm, I guess I don't properly understand the reasoning/use case then. Why would we prevent sending updates to the demo server, when it's (seemingly?) OK to send them? I thought we either shouldn't do it at all or unconditionally allow it.

Reply:
That's a totally fair point ... so why not prevent that upload in demo mode altogether?

Reply:
I pushed the implementation you suggested. If a user is connected to a server, there is an error sending some information, the user turns on demo mode and click on retry, the info is send to the demo server.